Volume/Views/MainView and Tabs/BookmarksView.swift

```diff
@@ -97,31 +105,83 @@ struct BookmarksView: View {
     // MARK: - Sections
 
     private var flyerContent: some View {
-        Group {
-            if viewModel.hasSavedFlyers {
-                switch viewModel.flyers {
-                case .none:
-                    ForEach(0..<4) { _ in
-                        FlyerCellPast.Skeleton()
-                            .padding(.bottom, 16)
-                    }
-                case .some(let flyers):
-                    ForEach(flyers) { flyer in
-                        if let urlString = flyer.imageUrl?.absoluteString {
-                            FlyerCellPast(
-                                flyer: flyer,
-                                navigationSource: .bookmarkFlyers,
-                                urlImageModel: URLImageModel(urlString: urlString),
-                                viewModel: FlyersView.ViewModel()
-                            )
+        VStack {
+            flyerSection(.upcoming)
+            flyerSection(.past)
+        }
+    }
+
+    private func flyerSection(_ flyerSection: FlyerSection) -> some View {
+        Section {
+            let selectedFlyers = flyerSection == .upcoming ? viewModel.upcomingFlyers : viewModel.pastFlyers
```

Comment:
This was a little weird to read at first glance, maybe some parenthesis would be goood